srcStatic/GameModules/DllModule.cpp

```diff
@@ -18,4 +19,56 @@ void DllModule::LoadModuleDll(const std::string& dllPath)
 	}
 
 	this->moduleDllHandle = dllHandle;
+
+	DetectExportedModuleFunctions();
+}
+
+void DllModule::DetectExportedModuleFunctions()
+{
+	loadModuleFunctionIni = (LoadModuleFunctionIni)GetProcAddress(moduleDllHandle, "InitMod");
+	if (loadModuleFunctionIni == nullptr) {
+		loadModuleFunctionConsole = (LoadModuleFunctionConsole)GetProcAddress(moduleDllHandle, "mod_init");
+	}
+
+	unloadModuleFunction = (UnloadModuleFunction)GetProcAddress(moduleDllHandle, "DestroyMod");
+	if (unloadModuleFunction == nullptr) {
+		unloadModuleFunction = (UnloadModuleFunction)GetProcAddress(moduleDllHandle, "mod_destroy");
```

Comment:
These methods differ by the return type. The `DestroyMod` address should not be casted to a type with a non-void return type. It may cause random garbage values left in registers to be interpreted as a return type. This may fool calling code into thinking the module failed to unload correctly.